File: 프로그래머스/1/133499. 옹알이 （2）/옹알이 （2）.java
class Solution {
    public int solution(String[] babbling) {
        int answer = 0;
        String[] list = { "aya", "ye", "woo", "ma" };

        for (String s : babbling) {
            StringBuilder sb = new StringBuilder();
            int lastIndex = -1;
            for (int j = 0; j < s.length(); j++) {
                char c = s.charAt(j);
                sb.append(c);

                for (int i = 0; i < 4; i++) {
                    if (i != lastIndex && list[i].equals(sb.toString())) {
                        sb.setLength(0);
                        lastIndex = i;
                    }
                }
            }
            if (sb.toString().isBlank()) {
                answer++;
            }
        }

        return answer;
    }
}

File: 프로그래머스/1/133499. 옹알이 （2）/옹알이 （2）.py
def solution(babbling):
    answer = 0
    words = ["aya", "ye", "woo", "ma"]

    for b in babbling:
        prev = ""
        while b:
            found = False
            for w in words:
                if b.startswith(w) and w != prev:
                    b = b[len(w):]
                    prev = w
                    found = True
                    break

            if not found:
                break

        if not b:
            answer += 1

    return answer